packages/@react-aria/calendar/src/useCalendarGrid.ts

```diff
@@ -137,13 +152,14 @@ export function useCalendarGrid(props: AriaCalendarGridProps, state: CalendarSta
   let dayFormatter = useDateFormatter({weekday: props.weekdayStyle || 'narrow', timeZone: state.timeZone});
   let {locale} = useLocale();
   let weekDays = useMemo(() => {
-    let weekStart = startOfWeek(today(state.timeZone), locale);
+    let offset = firstDayOfWeek ? (DAY_MAP[firstDayOfWeek] - getWeekStart(locale) + 7) % 7 : 0;
+    let weekStart = startOfWeek(today(state.timeZone), locale).add({days: offset});
```

Comment:
Seems like we need this logic in a few places. Can we add it to the `startOfWeek` function as an optional third argument? For example:

```js
startOfWeek(date, locale, 'mon')
```

Or actually in the case of a fixed week start we don't even need the locale. That might simplify the logic too so you don't have two offsets. An alternative API could be an options object. Not sure

```js
startOfWeek(date, {firstDayOfWeek: 'mon'})
```

Ideally the other functions that take a locale like `getDayOfWeek` and `getWeeksInMonth` would also take this option.

Reply:
Updated to use `startOfWeek(date, locale, 'mon')` to prevent breaking changes to `startOfWeek`